lib/libfrr.c

```diff
@@ -261,6 +261,35 @@ bool frr_zclient_addr(struct sockaddr_storage *sa, socklen_t *sa_len,
 
 static struct frr_daemon_info *di = NULL;
 
+void frr_process_guard(void)
+{
```

Comment:
1) I think it will be better to have a return value for this function and then do exit(1) in the actual main function based on the return value. That way it will be more clear to anyone who is following the code. Just think and change if needed.

2) Do you think this code needs to be covered with a lock to avoid multiple daemons running in any corner case ?

Reply:
1. If you looks into the file "lib/libfrr.c" almost all the functions calling syscalls/linux lib calls are returning void. If i decide to change this function, logically i should be doing for all others. For this is might get another comment. That’s is the reason i tried to retain the present coding behaviour.
2. Logically it would be good to have a lock though it is unlikely to happen in the real world as we need to run FRR or its daemons from two different terminals and at the same time. Moreover it would be difficult to implement this as each daemon will have its own copy.